src/adapters/IDB.js

```diff
@@ -340,12 +348,9 @@ export default class IDB extends BaseAdapter {
       });
     })
     .then((results) => {
-      // The resulting list of records is filtered and sorted.
-      const remainingFilters = {...filters};
-      // If `indexField` was used already, don't filter again.
-      delete remainingFilters[indexField];
+      // The resulting list of records is sorted.
       // XXX: with some efforts, this could be fully implemented using IDB API.
-      return reduceRecords(remainingFilters, params.order, results);
```

Comment:
As `reduceRecords` isn't used anymore, it should be removed (check for tests too)